src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+  * @param mean_truth_block_length the mean phased block length in the truth.
+  * @param median_truth_block_length the median phased block length in the truth.
+  * @param stddev_truth_block_length the standard deviation of the phased block length in the truth.
+  * @param n50_truth_block_length the N50 of the phased block length in the truth.
+  * @param n90_truth_block_length the N90 of the phased block length in the callset.
+  * @param l50_truth_block_length the L50 of the phased block length in the callset.
+  */
+case class AssessPhasingMetric
+(
+  var num_called: Long = 0,
+  var num_phased: Long = 0,
+  var num_truth: Long = 0,
+  var num_truth_phased: Long = 0,
+  var num_called_with_truth_phased: Long = 0,
+  var num_phased_with_truth_phased: Long = 0,
+  var num_truth_phased_in_called_block: Long = 0,
+  var num_both_phased_in_called_block: Long = 0,
+  var num_short_switch_errors: Long = 0,
+  var num_long_switch_errors: Long = 0,
+  var num_long_switch_sites: Long = 0,
+  var num_illumina_point_switch_errors: Long = 0,
+  var num_illumina_long_switch_errors: Long = 0,
+  var num_illumina_long_switch_sites: Long = 0,
+  var frac_phased: Double = 0,
+  var frac_phased_with_truth_phased: Double = 0,
+  var frac_truth_phased_in_called_block: Double = 0,
+  var frac_phased_with_truth_phased_in_called_block: Double = 0,
+  var mean_called_block_length: Double = 0,
+  var median_called_block_length: Double = 0,
+  var stddev_called_block_length: Double = 0,
+  var n50_called_block_length: Double = 0,
+  var n90_called_block_length: Double = 0,
+  var l50_called_block_length: Double = 0,
+  var mean_truth_block_length: Double = 0,
+  var median_truth_block_length: Double = 0,
+  var stddev_truth_block_length: Double = 0,
+  var n50_truth_block_length: Double = 0,
+  var n90_truth_block_length: Double = 0,
+  var l50_truth_block_length: Double = 0
+) extends Metric {
+  private def divide(a: Double, b: Double): Double = {
+    if (b == 0) 0
+    else a / b
+  }
+  def finalizeMetric(): this.type = {
+    this.frac_phased                                   = this.divide(this.num_phased, this.num_called)
+    this.frac_phased_with_truth_phased                 = this.divide(this.num_phased_with_truth_phased, this.num_called_with_truth_phased)
+    this.frac_truth_phased_in_called_block             = this.divide(this.num_truth_phased_in_called_block, this.num_truth_phased)
+    this.frac_phased_with_truth_phased_in_called_block = this.divide(this.num_both_phased_in_called_block, this.num_truth_phased_in_called_block)
+    this
+  }
+}
+
+/** The cigar Elements */
+private object CigarType extends Enumeration {
+  type CigarType = Value
+  val Match,      // 0
+  Mismatch,       // 1
+  Truth,          // 2
+  Call,           // 3
+  TruthEnd,       // 4
+  CallEnd,        // 5
+  BothEnd = Value // 6
+}
+
+private[vcf] object Cigar {
+  import CigarType._
+
+  type Cigar = Seq[CigarType]
```

Comment:
You have a bunch of methods in the object here that take a single `Cigar` as their first (sometimes only) parameter.  I think you'd be better off implementing a trivial Cigar _class_ that has a single member that is a `Seq[CigarType]` and has those methods on it.